File: cpp/2325.cpp
class Solution {
public:
    string decodeMessage(string& key, string& message) {
        char tf[26] = {0};
        int n = key.length();
        char s = 'a';
        for (int i = 0; i < n; i++) {
            int t = key[i] - 'a';
            if (t > -1 && tf[t] == 0)
                tf[t] = s++;
        }
        n = message.length();
        for (int i = 0; i < n; i++)
            if (message[i] != ' ')
                message[i] = tf[message[i] - 'a'];
        return message;
    }
};

File: python3/2325.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        tf = [0] * 26
        s = 97
        res = []
        for i in key:
            if i != " " and tf[ord(i) - 97] == 0:
                tf[ord(i) - 97] = chr(s)
                s += 1
        for i in message:
            if i != " ":
                res.append(tf[ord(i) - 97])
            else:
                res.append(" ")
        return "".join(res)
